packages/@ember/-internals/glimmer/tests/unit/runtime-resolver-cache-test.js

```diff
@@ -43,94 +33,91 @@ moduleFor(
       );
 
       this.assertText('foo-bar helper!');
-      state = this.expectCacheChanges(
-        { helperDefinitionCount: 1 },
-        state,
+      this.expectCacheChanges(
+        {
+          helperDefinitionCount: 1,
+          // from this.render
+          templateCacheMisses: 1,
+        },
         'calculate foo-bar helper only'
       );
 
       // show component-two for the first time
       runTask(() => set(this.context, 'cond', false));
 
       this.assertText('baz-qux helper!');
-      state = this.expectCacheChanges(
+      this.expectCacheChanges(
         {
           helperDefinitionCount: 1,
         },
-        state,
         'calculate baz-qux helper, misses cache'
       );
 
       // show foo-bar again
       runTask(() => set(this.context, 'cond', true));
 
       this.assertText('foo-bar helper!');
-      state = this.expectCacheChanges({}, state, 'toggle back to foo-bar cache hit');
+      this.expectCacheChanges({}, 'toggle back to foo-bar cache hit');
 
       // show baz-qux again
       runTask(() => set(this.context, 'cond', false));
 
       this.assertText('baz-qux helper!');
-      state = this.expectCacheChanges({}, state, 'toggle back to baz-qux cache hit');
+      this.expectCacheChanges({}, 'toggle back to baz-qux cache hit');
     }
 
-    '@test a component definition is only generated once'(assert) {
+    '@test a component definition is only generated once'() {
       // static layout
       this.registerComponent('component-one', { template: 'One' });
       this.registerComponent('component-two', {
         ComponentClass: Component.extend(),
         template: 'Two',
       });
 
-      // assert precondition
-      let state = this.getCacheCounters();
-      assert.deepEqual(
-        state,
-        {
-          templateCacheHits: 0,
-          templateCacheMisses: 0,
-          componentDefinitionCount: 0,
-          helperDefinitionCount: 0,
-        },
-        'precondition'
-      );
+      // snapshot counters
+      this.getCacheCounters();
 
       // show component-one for the first time
       this.render(`{{component componentName}}`, {
         componentName: 'component-one',
       });
 
       this.assertText('One');
-      state = this.expectCacheChanges(
-        { componentDefinitionCount: 1 },
-        state,
+      this.expectCacheChanges(
+        {
+          componentDefinitionCount: 1,
+          // 1 from this.render, 1 from component-one
+          templateCacheMisses: 2,
```

Comment:
prior to this change, this path was only used for late bound templates that were cached because their lifetime wasn't in the DI system's container.

Reply:
Yep, thats correct. Are you saying this is bad/wrong/etc?